src/python/bindings.cpp

```diff
@@ -504,6 +523,12 @@ PYBIND11_MODULE(pyqmap, m) {
           "n_threads_heuristic", &cs::Configuration::nThreadsHeuristic,
           "Maximum number of threads used for the heuristic optimizer. "
           "Defaults to the number of available threads on the system.")
+      .def_readwrite("gate_set", &cs::Configuration::gateSet,
+                     "Gate Set to be used for the Synthesis. "
+                     "Defaults to {H, S, Sdg, X, Y, Z, CX}.")
+      .def_readwrite("delay_paulis", &cs::Configuration::delayPaulis,
+                     "Delay Pauli gates to the end of the circuit. "
+                     "Defaults to `false`.")
```

Comment:
I believe it would be good to add a statement here that comments on whether this preserves optimality.